llmfoundry/utils/builders.py

```diff
@@ -155,6 +159,35 @@ def build_icl_data_and_gauntlet(
     return icl_evaluators, logger_keys, eval_gauntlet_cb
 
 
+def build_composer_model(
+    name: str,
+    cfg: DictConfig,
+    tokenizer: PreTrainedTokenizerBase,
+    init_context: Optional[ContextManager] = None,
+) -> ComposerModel:
+    if init_context is None:
+        init_context = contextlib.nullcontext()
+
+    with init_context:
+        model = construct_from_registry(
+            name=name,
+            registry=registry.models,
+            pre_validation_function=ComposerModel,
+            post_validation_function=None,
+            kwargs={
+                'om_model_config': cfg,
+                'tokenizer': tokenizer
+            },
+        )
+
+    if cfg.get('master_weights_dtype') in ('bf16', 'bfloat16'):
```

Comment:
nit: pass in the `master_weights_dtype` as a direct arg to `build_composer_model`

Reply:
sure, i guess i can pop them off and start down the config improvement road